lib/widgets/actions.dart

```diff
@@ -31,6 +31,29 @@ abstract final class ZulipAction {
   /// for details on the UI feedback, see there.
   static Future<void> markNarrowAsRead(BuildContext context, Narrow narrow) async {
     final zulipLocalizations = ZulipLocalizations.of(context);
+    final store = PerAccountStoreWidget.of(context);
+
+    final unreadCount = switch (narrow) {
+      CombinedFeedNarrow() => store.unreads.countInCombinedFeedNarrow(),
+      ChannelNarrow(:final streamId) => store.unreads.countInChannelNarrow(streamId),
+      MentionsNarrow() => store.unreads.countInMentionsNarrow(),
+      KeywordSearchNarrow() => store.unreads.countInKeywordSearchNarrow(),
+      StarredMessagesNarrow() => store.unreads.countInStarredMessagesNarrow(),
+      _ => 0,
+    };
```

Comment:
```suggestion
    final unreadCount = store.unreads.countInNarrow(narrow);
```

right?

Reply:
Also, since `unreadCount` is only used for the confirmation dialog, let's skip the work of computing it except if we're going to show the confirmation dialog.